crates/engine/tree/src/tree/payload_processor.rs

```diff
@@ -0,0 +1,230 @@
+//! Entrypoint for payload processing.
+
+use crate::tree::{
+    cached_state::{CachedStateMetrics, ProviderCaches},
+    root::{SparseTrieUpdate, StateRootConfig, StateRootTaskMetrics},
+    StateProviderBuilder,
+};
+use alloy_consensus::transaction::Recovered;
+use reth_primitives_traits::{header::SealedHeaderFor, NodePrimitives};
+use reth_provider::{
+    BlockReader, DatabaseProviderFactory, HashedPostStateProvider, StateCommitmentProvider,
+    StateProviderFactory, StateReader, StateRootProvider,
+};
+use reth_revm::cancelled::ManualCancel;
+use reth_workload_executor::WorkloadExecutor;
+use std::{
+    collections::VecDeque,
+    sync::{
+        mpsc,
+        mpsc::{Receiver, Sender},
+    },
+};
+
+/// Entrypoint for executing the payload.
+pub struct PayloadProcessor {
+    executor: WorkloadExecutor,
+    // TODO move all the caching stuff in here
+}
+
+impl PayloadProcessor {
+    /// Executes the payload based on the configured settings.
+    pub fn execute(&self) {
+        // TODO helpers for executing in sync?
+    }
+
+    /// Spawns all background tasks and returns a handle connected to the tasks.
+    ///
+    /// - Transaction prewarming task
+    /// - State root task
+    /// - Sparse trie task
+    ///
+    /// # Transaction prewarming task
+    ///
+    /// Responsible for feeding state updates to the state root task.
+    ///
+    /// This task runs until:
+    ///  - externally cancelled (e.g. sequential block execution is complete)
+    ///  - all transaction have been processed
+    ///
+    /// ## State root task
+    ///
+    /// Responsible for preparing sparse trie messages for the sparse trie task.
```

Comment:
i see, so state root task will only spawn multiproofs and send results back to `PayloadProcessor`, but `PayloadProcessor` itself will send the state update + multiproof to sparse trie task?

Reply:
> send results back to PayloadProcessor

send back to itself. PayloadProcessor is just some helper that spawns these 3 tasks